proposals/0000-swift-settings.md

```diff
@@ -0,0 +1,446 @@
+in `Package.swift` files would always need to be namespaced. The authors think
+that this is an acceptable trade-off since:
+
+* The `swiftpm` codebase is one project out of many and can just namespace as
+  appropriate.
+* `Package.swift` files really shouldn't need `#SwiftSettings` since they are
+  very self contained APIs.
+
+## ABI compatibility
+
+This proposal does not inherently break ABI. But it can break ABI if a command
+line option is enabled that causes the generated code's ABI to change. As part
+of enabling an option, one must consider such implications.
+
+## Implications on adoption
+
+Adopters of this proposal should be aware that while this feature does not
+inherently break ABI, enabling options using `#SwiftSettings` that break ABI can
+result in ABI breakage.
+
+## Future directions
+
+### Adding support for warningsAsErrors and strictConcurrency to allow for per-file warningsAsErrors based migration of large modules to strict concurrency
+
+Swift has taken the position that updating modules for strict concurrency should
+be accomplished by:
+
+1. Enabling strict concurrency on the entire module causing the compiler to emit
+   warnings.
+
+2. Updating the module incrementally until all of the warnings have been
+   eliminated from the module.
+
+3. Updating the language version of the module to swift 6 so from that point on
+   warnings will become errors to prevent backsliding.
+
+This creates developer friction when applied to large modules, since the size of
+the module makes it difficult to eliminate all of the warnings at once. This
+results in either the module being updated over time on main during which main
+is left in an intermediate, partially updated state or the module being updated
+on a branch that is merged once all updates have been completed. In the former
+case, it is easy to introduce new warnings as new code is added to the codebase
+since only warnings are being used. In the later case, the branch must be kept
+up to date in the face of changes being made to the codebase on main, forcing
+one to continually need to update the branch to resolve merge conflicts against
+main.
+
+In contrast by allowing for strict concurrency to be applied one file at a time,
+users can update code using a warningsAsErrors based migration path. This
+involves instead:
+
+1. Enabling strict concurrency and warningsAsErrors for strict concurrency on
+   the specific file.
+
+2. Fixing all of the errors in the file.
+
+3. Commiting the updated file into main.
+
+Since one is only updating one file at a time and using warningsAsErrors, the
+updates can be done incrementally on a per file basis, backsliding cannot occur
+in the file, and most importantly main is never in an intermediate, partially
+updated state. We think that this approach will make it significantly easier for
+larger modules to be migrated to strict concurrency.
+
+#### `strictConcurrency`
```

Comment:
I recommend omitting this section and the next one; since you've left `strictConcurrency` and `warningsAsErrors` as a future direction, it's not a good use of time for reviewers to comment on the details of how it would be accomplished in the review thread for this proposal.